src/librados/librados.cc

```diff
@@ -624,6 +624,18 @@ uint32_t librados::NObjectIteratorImpl::seek(uint32_t pos)
   return r;
 }
 
+void librados::NObjectIteratorImpl::set_filter(const bufferlist &bl)
+{
+  assert(ctx);
+  if (ctx->nlc) {
+    ctx->nlc->filter = bl;
+  }
+
+  if (ctx->lc) {
+    ctx->lc->filter = bl;
```

Comment:
Does this mean you are enhancing the semantics of the non-all-namespace over-wire operation?  What I did to maintain backwards compatibility when first adding the pgnls, variant is confusing.  It would be nice to phase pgls out once we know that pgls and pgnls OSDs don't have to interoperate.